src/main/kotlin/org/radarbase/redcap/managementportal/Subject.kt

```diff
@@ -31,19 +31,20 @@ import java.net.URL
 /**
  * Constructor.
  * @param subjectId [String] representing Management Portal Subject identifier
- * @param externalId [Integer] representing the REDCap Record identifier
- * @param externalLink [URL] pointing the REDCap integration form / instrument
+ * @param externalId [String] representing the REDCap Record identifier
+ * @param externalLink [String] pointing the REDCap integration form / instrument
  * @param project [Project] associated with the subject
  * @param attributes [Map] of key,value pairs
  */
 data class Subject(
     @JsonProperty("id") val mpId: Long?,
     @JsonProperty("login") val subjectId: String,
-    @JsonProperty("externalId") val externalId: Int? = null,
-    @JsonProperty("externalLink") val externalLink: URL? = null,
+    @JsonProperty("externalId") val externalId: String? = null,
```

Comment:
IIRC redcap ids are always Int. Any reason this was changed to string? I assume because in MP this is a string field?

Reply:
Yeah in MP, it's a string and there are issues when we pull the subjects from MP. It was working so far because in all the projects we use the integrator, we create all the subjects from REDCap. The issue is when it's a mix of user-created and REDCap-created (for example, in `STAGING_PROJECT`).